server/game/GameActions/DiscardStatusAction.ts

```diff
@@ -1,26 +1,15 @@
-import { CardGameAction, CardActionProperties} from './CardGameAction';
-import { CardTypes, Locations, EventNames } from '../Constants';
-import AbilityContext = require('../AbilityContext');
-import BaseCard = require('../basecard');
+import { TokenAction, TokenActionProperties} from './TokenAction';
+import { EventNames } from '../Constants';
 
-export interface DiscardStatusProperties extends CardActionProperties {
+export interface DiscardStatusProperties extends TokenActionProperties {
 }
 
-export class DiscardStatusAction extends CardGameAction {
+export class DiscardStatusAction extends TokenAction {
     name = 'discardStatus';
-    eventName = EventNames.OnCardStatusDiscarded;
+    eventName = EventNames.OnStatusTokenDiscarded;
     effect = 'discard {0}\'s status token';
-    targetType = [CardTypes.Character];
-
-    canAffect(card: BaseCard, context: AbilityContext): boolean {
-        if(card.location !== Locations.PlayArea || !card.isHonored && !card.isDishonored) {
-            return false;
-        }
-        return super.canAffect(card, context);
-    }
 
     eventHandler(event): void {
-        event.card.isHonored = false;
-        event.card.isDishonored = false;
+        event.token.card.makeOrdinary();
```

Comment:
This will (I think) still set as ordinary even if the token has changed, right?  Unless we set card as null when the token is removed from play (see comment below)

Reply:
Yes, you're right - see solution below